File: unit_4/w10d04/instructor_examples/dog.js
class Dog {
    constructor(name) {
        this.legs = 4
        this.name = name
    }

    bark() {
        console.log('bark!')
    }
    
    codeWord() {
        console.log('Blitzkrieg...')
    }
}

// create instance
const fido = new Dog('Fido')
// call bark
fido.bark()
// read name
console.log(fido.name)
// write name
fido.name = "Lucky"
console.log(fido.name)
// try to call secret method
fido.codeWord() // it runs...

File: unit_4/w10d04/instructor_examples/dog.rb
class Dog
    attr_accessor :name, :legs

    def initialize(name)
        @legs = 4
        @name = name
    end

    def bark
        puts "bark!"
    end

    private 

    def code_word
        puts "Blitzkrieg..."
    end

end


# create instance
fido = Dog.new("Fido")
# call bark
fido.bark
# read name
puts fido.name
# write name
fido.name = "Lucky"
puts fido.name
# try to call secret method
fido.code_word # it doesn't run!
